lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -70,41 +70,59 @@ public static GroupColumnIO getMapKeyValueColumn(GroupColumnIO groupColumnIO)
     }
 
     /* For backward-compatibility, the type of elements in LIST-annotated structures should always be determined by the following rules:
-     * 1. If the repeated field is not a group, then its type is the element type and elements are required.
-     * 2. If the repeated field is a group with multiple fields, then its type is the element type and elements are required.
-     * 3. If the repeated field is a group with one field and is named either array or uses the LIST-annotated group's name with _tuple appended then the repeated type is the element type and elements are required.
-     * 4. Otherwise, the repeated field's type is the element type with the repeated field's repetition.
      * https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists
      */
-    public static ColumnIO getArrayElementColumn(ColumnIO columnIO)
+    public static ColumnIO getArrayElementColumn(Type type, ColumnIO repeatedColumnIO)
     {
-        while (columnIO instanceof GroupColumnIO && !columnIO.getType().isRepetition(REPEATED)) {
```

Comment:
Are we missing recursive aspect here ? Previously it would work for nested type with more levels, how would it work in this case ?